src/components/shared/Fields/Input/HookForm/FormattedInputComponent.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
```

Comment:
In the Formik version, this lives inside `inputComponent`, but it's very messy in there so I've split this out into it's own component for use in the Hook Form version.